### ISSUE TITLE
thinning value in thin_draws is truncated if non-integer

### ISSUE THREAD

**Issue by n-kall** [stan-dev/posterior#328]
Currently the `thin` value given to `thin_draws` is passed through `as_one_integer` which uses `as.integer`. 
`as.integer` truncates rather than rounding.
This should probably be documented or perhaps changed.

```r
as.integer(1.9)
#> 1
```

**Comment by n-kall** [comment]
@avehtari suggested in #326 to allow non-integer thinning via rounding the indices generated. Perhaps `thin_draws` should allow this
```r
round(seq(1, 10, by = 1.5))
#> [1] 1 2 4 6 7 8 10
```

**Comment by n-kall** [comment]
@paul-buerkner do you think it is reasonable to change how the thinning value is interpreted (e.g. to make it interpret non-integer values directly) or would this be a potentially breaking change that would need to be optional?

**Comment by paul-buerkner** [comment]
Yes, I like the change. Do you have time to implement it?

**Comment by n-kall** [comment]
Yes, I can add it to #326

**Comment by n-kall** [comment]
Closed by #326
